tests/subsys/bluetooth/mesh/scheduler_model/message_validity/src/main.c

```diff
@@ -0,0 +1,477 @@
+}
+
+static void *setup(void)
+{
+	mod_elem_idx_prep();
+
+	/* 1st of Jan 2010 */
+	start_tm.tm_year = 110;
+	start_tm.tm_mon = 0;
+	start_tm.tm_mday = 1;
+	start_tm.tm_wday = 5;
+	start_tm.tm_hour = 0;
+	start_tm.tm_min = 0;
+	start_tm.tm_sec = 0;
+
+	/* all valid action values */
+	valid_entry.year = 0x63;
+	valid_entry.month = ANY_MONTH;
+	valid_entry.day = 5;
+	valid_entry.hour = 5;
+	valid_entry.minute = 5;
+	valid_entry.second = 30;
+	valid_entry.day_of_week = ANY_DAY_OF_WEEK;
+	valid_entry.action = BT_MESH_SCHEDULER_TURN_ON;
+	valid_entry.transition_time = model_transition_encode(2500);
+	valid_entry.scene_number = 1;
+
+	zassert_not_null(_bt_mesh_scheduler_srv_cb.init, "Server init cb is null");
+	zassert_not_null(_bt_mesh_scheduler_cli_cb.init, "Client init cb is null");
+
+	_bt_mesh_scheduler_srv_cb.init(sched_mod1_elem);
+	_bt_mesh_scheduler_cli_cb.init(sched_mod2_elem);
+
+	return NULL;
+}
+
+static void run_before(void *f)
+{
+	k_sem_reset(&action_set_cb_called);
+	k_sem_reset(&action_status_rcvd);
+	k_sem_reset(&status_rcvd);
+	k_sem_reset(&status_send);
+}
+
+static void teardown(void *f)
+{
+	zassert_not_null(_bt_mesh_scheduler_srv_cb.reset, "Server reset cb is null");
+	zassert_not_null(_bt_mesh_scheduler_cli_cb.reset, "Client reset cb is null");
+
+	_bt_mesh_scheduler_srv_cb.reset(sched_mod1_elem);
+	_bt_mesh_scheduler_cli_cb.reset(sched_mod2_elem);
+}
+
+ZTEST_SUITE(scheduler_message_validity_test, NULL, setup, run_before, NULL, teardown);
+
+ZTEST(scheduler_message_validity_test, test_cli_set_prohibited)
+{
+	/** (1) Send a SET and SET_UNACK message with a prohibited schedule entry, one for each
+	 * possible prohibited value according to the spec.
+	 * and (2) verify that the client returns with error (no message is sent to server).
+	 */
+
+	struct bt_mesh_schedule_entry test_entry[6];
+
+	for (size_t i = 0; i < ARRAY_SIZE(test_entry); i++) {
+		test_entry[i] = valid_entry;
+	}
+
+	/* prohibited struct member values */
+	test_entry[0].action = 4;
+	test_entry[1].year = 0x65;
+	test_entry[2].day = 0x21;
+	test_entry[3].hour = 0x1a;
+	test_entry[4].minute = 0x40;
+	test_entry[5].second = 0x40;
+
+	for (int i = 0; i < ARRAY_SIZE(test_entry); i++) {
+		cli_set_msg_send_assert(&test_entry[i], test_idx, -EINVAL);
+	}
+}
+
+ZTEST(scheduler_message_validity_test, test_cli_set_valid)
+{
+	/** With a valid test action entry:
+	 * (1) Verify that the client accepts the valid entry and proceeds to send the set-message(s),
+	 * both acknowledged and unacknowledged.
+	 */
+
+	cli_set_msg_send_assert(&valid_entry, test_idx, 0);
+}
+
+ZTEST(scheduler_message_validity_test, test_cli_get_action_prohibited)
+{
+	/** (1) Send a ACTION GET message with a prohibited schedule action index count.
+	 *  (2) Verify that the client deny the prohibited index value (no message is sent)
+	 */
+
+	test_idx = BT_MESH_SCHEDULER_ACTION_ENTRY_COUNT;
+	zassert_equal(bt_mesh_scheduler_cli_action_get(&sched_cli, NULL, test_idx, NULL), -EINVAL);
```

Comment:
Use the ztest mocking module here to ensure that the msg_send mock is _not_ called.